src/app/features/filter-bar/filter-bar.component.html

```diff
@@ -0,0 +1,47 @@
+<div class="filters-wrapper">
+    <mat-form-field appearance="fill" style="background: white">
+        <mat-label>Date of birthday range</mat-label>
+        <mat-date-range-input [formGroup]="filters.byDate" [rangePicker]="picker">
+            <input matStartDate formControlName="start" placeholder="Start date" />
+            <input matEndDate formControlName="end" placeholder="End date" />
+        </mat-date-range-input>
+        <mat-hint>TYPE: MM/DD/YYYY – MM/DD/YYYY</mat-hint>
+        <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
+        <mat-date-range-picker #picker></mat-date-range-picker>
+    </mat-form-field>
+
+    <form [formGroup]="stateForm">
+        <input
+            type="text"
+            placeholder="States of USA"
+            matInput
+            formControlName="stateGroup"
+            required
+            [(ngModel)]="filters.byState"
+            [matAutocomplete]="autoGroup"
+        />
+        <mat-autocomplete #autoGroup="matAutocomplete">
+            <mat-optgroup *ngFor="let group of stateGroupOptions | async" [label]="group.letter">
+                <mat-option *ngFor="let name of group.names" [value]="name">
+                    {{ name }}
+                </mat-option>
+            </mat-optgroup>
+        </mat-autocomplete>
+    </form>
+
+    <div>
+        <input
+            type="text"
+            placeholder="Select gender"
+            matInput
+            [(ngModel)]="filters.byGender"
+            [matAutocomplete]="auto"
+        />
+        <mat-autocomplete #auto="matAutocomplete">
+            <mat-option *ngFor="let option of genderOptions; trackBy: genderTrackBy" [value]="option.id">
+                {{ option.name }}
+            </mat-option>
+        </mat-autocomplete>
+    </div>
+    <button (click)="apply.emit(filters)">APPLY</button>
```

Comment:
always add `type` for buttons because buttons without type have `type=submit` which may unexpectedly send form